src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,34 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * The parent class for Email, Address and phone.
+ */
+public class Contact {
+    public final String value;
```

Comment:
try to make it private. in general, start off with private members, then use protected or public only if absolutely necessary.